### PR TITLE
Added boilerplate for emoji plugin

### LINKED ISSUE #17376: Create `ckeditor5-emoji` package
## Provide a description of the task

We need to prepare an empty package to start working on a new feature.

### Definition of Done

* Based on an existing package, create a similar setup.
	* Example: https://github.com/ckeditor/ckeditor5/tree/master/packages/ckeditor5-adapter-ckfinder
* Create an empty `Emoji` plugin (it will be extended soon).
* Feature branch and a pull request should target the `#epic/ck/17361` branch.

### PR DESCRIPTION
**Pull request by przemyslaw-zan**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/contributing/git-commit-message-convention.html))

Internal: Added boilerplate for emoji plugin. Closes #17376.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Comment by pomek** [comment]
Please make CI green. Even with an empty package, it's possible.

File: packages/ckeditor5-emoji/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    "@ckeditor/ckeditor5-cloud-services": "43.3.0",
+    "@ckeditor/ckeditor5-dev-utils": "^45.0.0",
+    "@ckeditor/ckeditor5-easy-image": "43.3.0",
+    "@ckeditor/ckeditor5-editor-classic": "43.3.0",
+    "@ckeditor/ckeditor5-image": "43.3.0",
+    "@ckeditor/ckeditor5-theme-lark": "43.3.0",
+    "typescript": "5.0.4",
+    "webpack": "^5.94.0",
+    "webpack-cli": "^5.1.4"
+  },
```
Comment (psmyrek) [inline packages/ckeditor5-emoji/package.json:32]: Let's add dependencies when they are really needed.
Emoji feature will not require `@ckeditor/ckeditor5-cloud-services`.

File: packages/ckeditor5-emoji/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module special-characters
```
Comment (psmyrek) [inline packages/ckeditor5-emoji/src/index.ts:7]:
```suggestion
 * @module emoji
```

File: package.json [hunk 3]
```diff
@@ -40,6 +40,7 @@
     "@ckeditor/ckeditor5-editor-decoupled": "43.3.0",
     "@ckeditor/ckeditor5-editor-inline": "43.3.0",
     "@ckeditor/ckeditor5-editor-multi-root": "43.3.0",
+    "@ckeditor/ckeditor5-emoji": "43.3.0",
```
Comment (psmyrek) [inline package.json:43]: This version does not exist.

File: packages/ckeditor5-emoji/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Contributing
+========================================
```
Comment (psmyrek) [inline packages/ckeditor5-emoji/CONTRIBUTING.md:2]: I know that this `=` line does not match the header in most other packages, but this is so ugly...

```suggestion
Contributing
============
```

File: packages/ckeditor5-emoji/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
+CKEditor&nbsp;5 emoji feature
+===========================
```
Comment (psmyrek) [inline packages/ckeditor5-emoji/README.md:2]:
```suggestion
CKEditor&nbsp;5 emoji feature
=============================
```

File: packages/ckeditor5-emoji/ckeditor5-metadata.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+	"plugins": [
+		{
+			"name": "Emoji",
+			"className": "Emoji",
+			"description": "Allows for inserting emojis via a dropdown or by typing `:` followed by emoji name.",
+			"path": "src/emoji.js"
```
Comment (psmyrek) [inline packages/ckeditor5-emoji/ckeditor5-metadata.json:7]: Docs path is already mentioned in several places. Let's add it to the metadata as well so we don't forget about it.

```suggestion
			"path": "src/emoji.js",
			"docs": "features/emoji.html"
```

**Review by psmyrek — APPROVED** [review]
LGTM 🎉

**Pull request merged**